src/kademlia/dht_tracker.cpp

```diff
@@ -294,26 +251,25 @@ namespace libtorrent { namespace dht
 		, boost::function<void(item const&, bool)> cb
 		, std::string salt)
 	{
-		m_dht.get_item(key, salt, boost::bind(&get_mutable_item_callback, _1, _2, cb));
+		m_dht.get_item(key, salt, cb);
 	}
 
 	void dht_tracker::put_item(entry data
-		, boost::function<void()> cb)
+		, boost::function<void(int)> cb)
 	{
 		std::string flat_data;
 		bencode(std::back_inserter(flat_data), data);
 		sha1_hash target = item_target_id(
 			std::pair<char const*, int>(flat_data.c_str(), flat_data.size()));
 
-		m_dht.get_item(target, boost::bind(&put_immutable_item_callback
-			, _1, cb, data));
+		m_dht.put_item(target, data, cb);
```

Comment:
I think it actually related to the next comment - the dht_tracker is kinda redundant.
dht_tracker actually doesn't have any dht logic, for example it doesn't know traversal_algorithm, so I can't add the put_data logic in it. that said, node.cpp has to export a new interface for put.